src/Core/Form/ChoiceProvider/ProductTypeChoiceProvider.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\ChoiceProvider;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductType;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceAttributeProviderInterface;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ProductTypeChoiceProvider implements FormChoiceProviderInterface, FormChoiceAttributeProviderInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(
+        TranslatorInterface $translator
+    ) {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getChoicesAttributes()
+    {
+        return [
+            $this->trans('Standard product', 'Admin.Catalog.Feature') => [
```

Comment:
The index of this array is the list of translated strings?

Reply:
@jolelievre 
I think it should be the other way around, and you could have everything in one method, what about:

```
public function getAvailableProductTypes()
{
    return [
        [
            ProductType::TYPE_STANDARD => [
                'label' => $this->trans('Standard product', 'Admin.Catalog.Feature'),
                'desc' => $this->trans('A physical product that needs to be shipped.', 'Admin.Catalog.Feature')
         ]
    ];
}
```

and if you want you can transform this array in `getChoices` and `getChoicesAttributes`?

Reply:
The format of `getChoices` and `getChoicesAttributes` actually match the expected format in `ChoiceType` from Symfony, so I can't turn around the format.

I could add a `getAvailableProductTypes` method but it shouldn't be public since it will be of no use for both interfaces that this service implements, so at best it would be a private method that is used by both public methods to reformat into the same format they already do separately.

I'm not sure this brings that much value 🤔